From: Tetsu Yamamoto <tyamamot@redhat.com>
Date: Tue, 2 Sep 2008 16:06:06 -0400
Subject: [xen] ia64: kludge for XEN_GUEST_HANDLE_64
Message-id: 20080902200606.7876.44374.sendpatchset@pq0-1.lab.bos.redhat.com
O-Subject: [RHEL5.3 PATCH 4/5] xen-ia64: Kludge for XEN_GUEST_HANDLE_64
Bugzilla: 454711
RH-Acked-by: Bill Burns <bburns@redhat.com>

bz454711
# HG changeset patch
# User awilliam@xenbuild2.aw
# Date 1170026961 25200
# Node ID 8b37c7bba7e12d306e720718fe8e1e6d6dee28eb
# Parent  cf0ae8105ffe32acd5a587ab31e91610febef8a6
[IA64] Kludge for XEN_GUEST_HANDLE_64

This is ugly, but the include spaghetti around xen.h isn't giving
us the right defines.  This should be removed with the includes
are untangled.

Signed-off-by: Alex Williamson <alex.williamson@hp.com>

diff --git a/include/xen/interface/arch-ia64.h b/include/xen/interface/arch-ia64.h
index dce2f24..8dc5a40 100644
--- a/include/xen/interface/arch-ia64.h
+++ b/include/xen/interface/arch-ia64.h
@@ -18,6 +18,8 @@
 
 #define DEFINE_XEN_GUEST_HANDLE(name)   __DEFINE_XEN_GUEST_HANDLE(name, name)
 #define XEN_GUEST_HANDLE(name)          __guest_handle_ ## name
+#define XEN_GUEST_HANDLE_64(name)       XEN_GUEST_HANDLE(name)
+#define uint64_aligned_t                uint64_t
 #define set_xen_guest_handle(hnd, val)  do { (hnd).p = val; } while (0)
 #ifdef __XEN_TOOLS__
 #define get_xen_guest_handle(val, hnd)  do { val = (hnd).p; } while (0)
